.github/workflows/publish_pytorch_dev_docker.yml

```diff
@@ -50,6 +52,16 @@ jobs:
       # This step uses the `docker/build-push-action` action to build the image, based on your repository's `Dockerfile`. If the build succeeds, it pushes the image to GitHub Packages.
       # It uses the `context` parameter to define the build's context as the set of files located in the specified path. For more information, see "[Usage](https://github.com/docker/build-push-action#usage)" in the README of the `docker/build-push-action` repository.
       # It uses the `tags` and `labels` parameters to tag and label the image with the output from the "meta" step.
+          tags: |
+            type=raw,value={{date 'YYYYMMDD'}}
+            type=schedule,pattern={{date 'YYYYMMDD'}}
+            type=ref,event=branch
+          labels: |
+            org.opencontainers.image.title=PyTorch ROCm Dev Image
+            org.opencontainers.image.description=Nightly ROCm PyTorch Dev Docker for target ${{ matrix.targets.amdgpu_target }}
+            org.opencontainers.image.version=nightly
```

Comment:
This label should be used to have the actual version and not the _release type_ (which is _nightly_ here). What you want here is more likely`X.Y.ZrcYYYYMMDD` to have a version identifier that is compatible with the Python world. See also https://www.docker.com/blog/docker-best-practices-using-tags-and-labels-to-manage-docker-image-sprawl/.

Reply:
One thing to consider is that labels are part of the image and therefore cannot be changed. Thus if we want to promote an image as stable, we need to rebuild. The alternative can be to use tags and set tags as immutable (to disallow pushing to the same tag). Following the best practices guide is that you need to make sure you don't forget to push the tag if you push a new image.

@ScottTodd any preference? As we seem to also aim for reproducible builds, labels might be the way to go.

Reply:
@marbre @ScottTodd , pushed some changes for versioning.

Reply:
Labels as part of the image SGTM. We can have a separate process for building and publishing "stable" / "official" releases/images.